client/src/app/home/home.component.ts

```diff
@@ -7,8 +8,25 @@ import { Component, OnInit } from '@angular/core';
 })
 export class HomeComponent implements OnInit {
     // eslint-disable-next-line @typescript-eslint/no-empty-function
-    constructor() {}
+    alert:boolean ;
+    mapReader = null;
+    constructor(private parseMapService: ParseMapService) {}
 
     // eslint-disable-next-line @typescript-eslint/no-empty-function
     ngOnInit(): void {}
+    readFile(file: FileList): void {
+        const mapFile = file.item(0);
+        const tab = mapFile.name.split('.');
+        if (tab[1] === 'txt') {
+            this.alert = true;
+            mapFile.text().then( (data) => {
+                this.mapReader = data;
+                this.parseMapService.parseFile(this.mapReader);
+            });
+        } else {
+            this.alert =false ;
+            console.log('File type error');
+            }
```

Comment:
Codacy found an issue: [Expected indentation of 8 spaces but found 12.](https://app.codacy.com/gh/Ca-Cybersecurity-Robiot/robiot/pullRequest?prid=5943029)